### ISSUE TITLE
[MASSEMBLY-661] Assembly plugin looses permissions when using fileSets

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-assembly-plugin#821]
**[Eric Haszlakiewicz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=erh)** opened **[MASSEMBLY-661](https://issues.apache.org/jira/browse/MASSEMBLY-661?redirect=false)** and commented

When using an assembly descriptor that uses fileSets to include files the permissions of the files from the file system are not preserved in the tarballs.
This can be reproduced by creating a file with the executable bit set in the "src" directory, then trying to use the pre-defined "src" descriptorRef to create a tarball.  See the attached pom file for an example.

---

**Affects:** 2.4

**Attachments:**
- [pom.xml](https://issues.apache.org/jira/secure/attachment/12714305/pom.xml) (_1.79 kB_)

**Issue Links:**
- [MASSEMBLY-714](https://issues.apache.org/jira/browse/MASSEMBLY-714) Update version of plexus-archiver to 2.7.1
 (_**"depends upon"**_)

**Comment by jira-importer** [comment]
**[Eric Haszlakiewicz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=erh)** commented

Note that there are other issues that talk about similar problems, but not quite the same as what I'm talking about here.
MASSEMBLY-238 (refers to depdencySets instead of fileSets)
MASSEMBLY-379 (marked as a dup of 238)
MASSEMBLY-423 (talks about repeated use of permissions)

**Comment by jira-importer** [comment]
**[Kristian Rosenvold](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=krosenvold)** commented

I just tested this on the official maven distribution;

```
git clone https://git-wip-us.apache.org/repos/asf/maven.git
cd maven
mvm clean install
cd apache-maven/target
```

extract zip file and tar file, the files in the "bin" folder seem to have the x bit set properly ?

this **might** be os/java version/file system specific, so please test the same.

**Comment by jira-importer** [comment]
**[Eric Haszlakiewicz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=erh)** commented

This is specific to the version maven.  Maven 2.2.1 is listed as "an older, but stable version of Maven for projects that can't upgrade to Maven 3.x yet", and I have several projects that can't upgrade yet.

Some time between 2.2.1 and 3.0.5 this problem was fixed, but it wasn't applied to the 2.2 code base.

**Comment by jira-importer** [comment]
**[Kristian Rosenvold](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=krosenvold)** commented

Please verify that you are using fixed versions of the plugins in question; to my best knowledge this is /not/ functionality/bugs that would leak from maven core into the assembly plugin, even for version 2.2.1 (it might happen somewhere in the 2.0.X range, but not 2.2.1)

The only thing I can think of that is markedly different between 2.2.1 and any 3.X version is the default version of all the plugins. Please double-check that you're not being tricked by some plugin being in an older version when running 2.2.1

**Comment by jira-importer** [comment]
**[Kristian Rosenvold](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=krosenvold)** commented

You should, of course, also check that the files have the expected permissions on disk within the target folder. As long as this is the case, there should be nothing in maven core that affects this (other than the default version of plugins)

**Comment by jira-importer** [comment]
**[Kristian Rosenvold](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=krosenvold)** commented

Full file mode support is added with archiver 2.6.1.

**Comment by jira-importer** [comment]
**[Kristian Rosenvold](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=krosenvold)** commented

Fixed with archiver 2.6.X. Please note that prefixing may still loose atetributes
